pkg/promotion/runner/builtin/schemas/git-push-config.json

```diff
@@ -28,6 +28,11 @@
       "type": "string",
       "description": "The name of the Git provider to use. Currently 'azure', 'bitbucket', 'gitea', 'github', and 'gitlab' are supported. Kargo will try to infer the provider if it is not explicitly specified.",
       "enum": ["azure", "bitbucket", "gitea", "github", "gitlab"]
+    },
+    "force": {
+      "type": "boolean",
+      "description": "Whether to force push to the target branch, overwriting any existing history. This is useful for scenarios where you want to completely replace the branch content (e.g., pushing rendered manifests that don't depend on previous state). Use with caution as this will overwrite any commits that exist on the remote branch but not in your local branch. When true, the step will skip the pull/rebase operation to allow overwriting remote history.",
```

Comment:
Just a reminder to update this after we figure out what we're putting [here](https://github.com/akuity/kargo/pull/5239/files#r2443347028).

I also wouldn't mention the pull rebase because that's not an option that's exposed to the user. It's an internal implementation detail.